src/FileSystem/SourceFileFilter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\FileSystem;
+
+use function explode;
+use Infection\FileSystem\Finder\Iterator\RealPathFilterIterator;
+use Infection\TestFramework\Coverage\CoveredFileData;
+use Iterator;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * @internal
+ * @final
+ */
+class SourceFileFilter
+{
+    /**
+     * @var string[]
+     */
+    private $filters;
+
+    /**
+     * @return iterable<SplFileInfo>
+     */
+    public function __construct(string $filter)
+    {
+        $this->filters = array_filter(array_map(
+            'trim',
+            explode(',', $filter)
+        ));
```

Comment:
I understand one can argue that with a single combo of `array_filter(array_map())` it may not make sense to use any wrappers, even if the wrapper does the same `array_filter` and `array_map`. Conversely, I will tend to disagree if we're talking several `array_*` calls in one place. They're a DX disaster, and parsing them is not what one wants to spend their invaluable minutes on, at least as I see it.

To make a point against learning curve and vendor lock-in arguments, I invite you as a contributor. Let's see if I can change your mind, and if not from the start, at least we might learn if there's anything more to be done about that.